plugins/by-name/image/default.nix

```diff
@@ -45,143 +47,97 @@ in
+        {
+          markdown.enabled = true;
+          typst.enabled = true;
+          neorg.enabled = true;
+          syslang.enabled = true;
+          html.enabled = false;
+          css.enabled = false;
+        }
+        ''
+          Per-filetype integrations.
+        '';
+
+    max_width = defaultNullOpts.mkUnsignedInt null ''
+      Image maximum width.
+    '';
+
+    max_height = defaultNullOpts.mkUnsignedInt null ''
+      Image maximum height.
+    '';
+
+    max_width_window_percentage = defaultNullOpts.mkUnsignedInt 100 ''
       Image maximum width as a percentage of the window width.
     '';
 
-    maxHeightWindowPercentage = helpers.defaultNullOpts.mkUnsignedInt 50 ''
+    max_height_window_percentage = defaultNullOpts.mkUnsignedInt 50 ''
       Image maximum height as a percentage of the window height.
     '';
 
-    windowOverlapClearEnabled = helpers.defaultNullOpts.mkBool false ''
+    window_overlap_clear_enabled = defaultNullOpts.mkBool false ''
       Toggles images when windows are overlapped.
     '';
 
-    windowOverlapClearFtIgnore =
-      helpers.defaultNullOpts.mkListOf types.str
+    window_overlap_clear_ft_ignore =
+      defaultNullOpts.mkListOf types.str
         [
           "cmp_menu"
           "cmp_docs"
-          ""
+          "snacks_notif"
+          "scrollview"
+          "scrollview_sign"
         ]
         ''
           Toggles images when windows are overlapped.
         '';
 
-    editorOnlyRenderWhenFocused = helpers.defaultNullOpts.mkBool false ''
+    editor_only_render_when_focused = defaultNullOpts.mkBool false ''
       Auto show/hide images when the editor gains/looses focus.
     '';
 
-    tmuxShowOnlyInActiveWindow = helpers.defaultNullOpts.mkBool false ''
+    tmux_show_only_in_active_window = defaultNullOpts.mkBool false ''
       Auto show/hide images in the correct Tmux window (needs visual-activity off).
     '';
 
-    hijackFilePatterns =
-      helpers.defaultNullOpts.mkListOf types.str
+    hijack_file_patterns =
+      defaultNullOpts.mkListOf types.str
         [
           "*.png"
           "*.jpg"
           "*.jpeg"
           "*.gif"
           "*.webp"
+          "*.avif"
         ]
         ''
           Render image files as images when opened.
         '';
   };
 
-  config = mkIf cfg.enable {
-    extraPlugins = [ cfg.package ];
-    extraLuaPackages = ps: [ ps.magick ];
+  settingsExample = {
+    backend = "kitty";
+    max_width = 100;
+    max_height = 12;
+    max_height_window_percentage.__raw = "math.huge";
+    max_width_window_percentage.__raw = "math.huge";
+    window_overlap_clear_enabled = true;
+    window_overlap_clear_ft_ignore = [
+      "cmp_menu"
+      "cmp_docs"
+      ""
+    ];
+  };
 
+  extraConfig = cfg: {
     extraPackages = [
       # In theory, we could remove that if the user explicitly disables `downloadRemoteImages` for
       # all integrations but shipping `curl` is not too heavy.
       cfg.curlPackage
```

Comment:
Do we want to address this old TODO comment?

Reply:
I don't fully understand, but I'll trust your decision.

Reply:
Maybe if it's something we don't want to do, we should update/remove the comment?